wordpress-mobile/WordPress-Android
# Plans on JP: Checkout Success

Updates checkout success screen to new design, and removes the old one 

Fixes: #18422 

| Before | After | Dark Mode|
|--------|--------|--------|
|![Screenshot_20230522_125624](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/addd7aeb-2439-4459-84af-16dd9fefd4e7)|![Screenshot_20230526_094453](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/0ad56401-00aa-48d0-95a2-bc0db3225d39)|![Screenshot_20230526_094507](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/2375c39d-fbcd-46e1-b63d-df14cdf5d520)|


To test:

Feature Flag
- Launch Jetpack app
- Go to **App Settings** (Tap on Avatar at the top right hand corner on My Site -> Find App Settings)
- Tap **Debug Settings**
- Find **dashboard_card_free_to_paid_plans** [under Remote features](https://github.com/wordpress-mobile/WordPress-Android/pull/18418)
- Enable the flag, and restart the app (scroll down)

Store sandbox
- Connect android emulator/device to sandbox paqN3M-5m-p2
- Use sandbox on Jetpack app to test for free with test cards pc8HXX-14D-p2

Success screen
- Verify **plans card** appears as shown above on dashboard
- Tap on **plans card**
- Verify it navigates to **Register domain** screen, and domains show label **Free for the first year**
- Select a domain, and Tap on **Select domain** button
- Verify that it goes to **Plans page** on the web
- Tap on Upgrade button of  a Personal or Premium Plan
- Verify it takes to checkout screen
- Use the test card numbers to purchase
- Verify the Success screen is shown as in after image above
- Verify the screen in landscape and dark mode as well
- Tap on Done button, return to dashboard
- If the app is connected to store sandbox, could verify that plans card doesn't show anymore
- Otherwise, go to store admin, change to sandbox (if Live), find the site, and verify the plan subscription is added


>**Note**
> - The url and display for plans page could change [based on this Calypso PR](https://github.com/Automattic/wp-calypso/pull/77029)


## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Unit tests and manual tests

3. What automated tests I added (or what prevented me from doing so)
Updated and added unit tests

PR submission checklist:



UI Changes testing checklist:
